publishers/README.md

```diff
@@ -0,0 +1,22 @@
+# Publishers
+
+This directory is the home for all published KRM functions metadata. Publishers are split into
+four categories:
+
+- communities
+- companies
+- github-orgs
+- individuals
```

Comment:
I'm inclined to agree that we can remove this layer. It's probably not important to distinguish between different types of contributors. Instead of this layer here, maybe each publisher can have their own README describing what they are. 

FYI @mengqiy

Reply:
The original intend to have this layer to enforce different CI check when a publisher publish a KRM function. For example, when publishing to a company directory, we may want to check the commit is made by someone with their company's email. When publishing for a GH org, we may want to check the commit is made by someone within the org.
If you don't think this is necessary, I'm OK to remove this layer.

Reply:
I'm not sure whether it is necessary yet, and if it is, I think we can handle it by introducing a configuration file that the function owner can customize to activate the checks appropriate for their org. So I'm not worried about leaving the layer out.

Reply:
> For example, when publishing to a company directory, we may want to check the commit is made by someone with their company's email. When publishing for a GH org, we may want to check the commit is made by someone within the org.

Since the commit would have to be approved by someone in the publisher's OWNERS file, I am also OK to remove the layer and agree with @KnVerey that we can find a way to introduce the check if/when we find it necessary.